### PR TITLE
enable explicitApi, cleanup exposed internal code

### PR DESCRIPTION
**Pull request by gabrielittner**
The reified inline methods now have a non inline overload that takes the class as parameter. This way we can completely hide the internal lists of builders

### CODE REVIEW

File: dsl/src/commonMain/kotlin/com/freeletics/flowredux/dsl/StoreWideBuilderBlock.kt [hunk 1]
```diff
@@ -13,7 +13,7 @@ import com.freeletics.flowredux.dsl.internal.Action
  * That is not that nice as it all subclasses are implementation detail heavy.
  * There is no need to have a hundreds of lines of code in one file just to have sealed classes.
  */
-abstract class StoreWideBuilderBlock<S, A> internal constructor() {
+public abstract class StoreWideBuilderBlock<S, A> internal constructor() {
```
Comment (gabrielittner) [inline dsl/src/commonMain/kotlin/com/freeletics/flowredux/dsl/StoreWideBuilderBlock.kt:16]: Should we remove the base class as long as it only has one usage?

Comment (sockeqwe) [reply dsl/src/commonMain/kotlin/com/freeletics/flowredux/dsl/StoreWideBuilderBlock.kt:16]: Yea probably 👍

**Comment by sockeqwe** [comment]
General question: 
don't you get now at least 2 autocomplete options for in the IDE? One fore `inState<Foo> { ... }` one for `inState(Foo::class) { ... } `?

I would prefer to not have these autocomplete overhead ...

**Comment by gabrielittner** [comment]
Maybe we can give those methods a different name? So that it at least doesn't appear when you type the correct first letter.

**Comment by sockeqwe** [comment]
Actual, is it so bad to expose
`lists of builders` ? Then people could expand DSL if they need to with their own functionality (more than just reusing existing DSL blocks)

**Comment by gabrielittner** [comment]
It's also requring the `OnActionInStateSideEffectBuilder` to be public, not just the builders list. If it were just that (or if we want to promote extensibility) I'd add a public`addBuilder` method

**Review by sockeqwe — APPROVED** [review]
👍

**Pull request merged**